inc/queuedmail.class.php

```diff
@@ -459,6 +459,25 @@ function sendMailById($ID) {
 
          if (!$mmail->Send()) {
             Session::addMessageAfterRedirect($messageerror."<br>".$mmail->ErrorInfo, true);
+
+            if (isset($CFG_GLPI['smtp_max_retries'])) {
+               $retries = $CFG_GLPI['smtp_max_retries'] - $this->fields['sent_try'];
+               Toolbox::logInFile("mail",
+                                sprintf(__('%1$s. Message: %2$s, Error: %3$s'),
+                                         sprintf(__('Warning: an email was undeliverable to %s with %d retries remaining'),
+                                                 $this->fields['recipient'], $retries),
+                                         $this->fields['name'],
+                                         $mmail->ErrorInfo."\n"));
+
+               if ($retries <= 0) {
+                  Toolbox::logInFile("mail",
```

Comment:
I went through most of the mailing code and didn't see anywhere that it logs for mail errors. I like the mail-error log idea.